{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PathFind scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction\n",
    "\n",
    "A series of scripts were developed so that users can access the results of the Sanger Pathogen Informatics analysis pipelines. These are referred to as the **pathfind** or **pf** scripts.\n",
    "\n",
    "\n",
    "| PathFind (pf) script | description                                                                               |\n",
    "| :-:        | ---                                                                                                 |\n",
    "| **status**     | returns the pipeline status (Running, Done or Failed) for each of the lanes                     |  \n",
    "| **data **      | finds the location on disk of sequence data tracked by the Pathogen Informatics pipelines       |\n",
    "| **info**       | returns metadata allowing you to match up internal sample identifiers with supplier identifiers |\n",
    "| **accession**  | returns the sample and raw data accession numbers for the sequence data                         |\n",
    "| **qc**         | returns the location of the Kraken report(s) generated by the QC pipeline                       | \n",
    "| **map**        | returns the location of the BAM files produced by the mapping pipeline                          | \n",
    "| **snp**        | returns the location of the VCF files produced by the SNP calling pipeline                      | \n",
    "| **assembly**   | returns the location of the contig FASTA files produced by the assembly pipeline                | \n",
    "| **annotation** | returns the location of the GFF files produced by the annotation pipeline                       | \n",
    "| **rnaseq**     | returns the location of the expression count files produced by the RNA-Seq pipeline             |\n",
    "| **ref**        | finds the location of reference files on disk                                                   |\n",
    "\n",
    "The source code for the pf script Perl module can be found on the [sanger-pathogens Git repository](https://github.com/sanger-pathogens) as [Bio-Path-Find](https://github.com/sanger-pathogens/Bio-Path-Find)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Learning outcomes\n",
    "\n",
    "By the end of this tutorial you can expect to be able to:\n",
    "\n",
    "  * Find the pipeline status for your lane(s) using the pf scripts\n",
    "  * Find the data for your lane(s) using the pf scripts\n",
    "  * Find the quality control (QC) results for your lane(s) using the pf scripts\n",
    "  * Find the analysis pipeline results for your lane(s) using the pf scripts\n",
    "  * Find a reference using the pf scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Tutorial sections\n",
    "\n",
    "  * [Introducing the pf scripts](intro.ipynb)\n",
    "  * [Pipeline status](status.ipynb)\n",
    "  * [Finding your data](data.ipynb)\n",
    "  * [Sample information and accessions](metadata.ipynb)\n",
    "  * [Quality control](qc.ipynb)\n",
    "  * [Analysis pipeline results](pipeline-results.ipynb)\n",
    "  * [Finding a reference](reference.ipynb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Authors\n",
    "\n",
    "This tutorial was created by [Victoria Offord](https://github.com/vaofford)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Running the commands from this tutorial\n",
    "\n",
    "You can run the commands in this tutorial either directly from the Jupyter notebook (if using Jupyter), or by typing the commands in your terminal window.\n",
    "\n",
    "### Running commands on Jupyter\n",
    "\n",
    "If you are using Jupyter, command cells (like the one below) can be run by selecting the cell and clicking _Cell -> Run_ from the menu above or using _Ctrl Enter_ to run the command. Let's give this a try by printing our working directory using the `pwd` command and listing the files within it. Run the commands in the two cells below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ls -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running commands in the terminal\n",
    "\n",
    "You can also follow this tutorial by typing all the commands you see into a terminal window. This is similar to the \"Command Prompt\" window on MS Windows systems, which allows the user to type DOS commands to manage files.\n",
    "\n",
    "To get started, select the cell below with the mouse and then either press control and enter or choose _Cell -> Run_ in the menu at the top of the page."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "echo cd $PWD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Open a new terminal on your computer and type the command that was output by the previous cell followed by the enter key. The command will look similar to this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cd /home/manager/pathogen-informatics-training/Notebooks/RNA-Seq/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now you can follow the instructions in the tutorial from here."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prerequisites\n",
    "\n",
    "This tutorial assumes that you have [Bio-Path-Find](https://github.com/sanger-pathogens/Bio-Path-Find) installed on your computer. For installation instructions please see [Bio-Path-Find](https://github.com/sanger-pathogens/Bio-Path-Find).\n",
    "\n",
    "To check that you have installed Bio-Path-Find correctly, you can run the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pf -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This should return the help message for the PathFind (pf) scripts."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Let's get started!\n",
    "\n",
    "To get started with the tutorial, head to the first section: [Introducing the pf scripts](intro.ipynb)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
