{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA-Seq pipeline results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction\n",
    "\n",
    "When your sample data is in the Pathogen Informatics databases, it becomes available to the automated analysis pipelines. After the analysis pipelines have been requested and run, you can use the `pf` scripts to return the results of each of the automated analysis pipelines.\n",
    "\n",
    "The [RNA-Seq expression pipeline](http://mediawiki.internal.sanger.ac.uk/index.php/Pathogen_Informatics_RNA-Seq_Expression_Pipeline) maps your raw sequence reads to a reference and counts the number of reads associated with each gene. We can use `pf rnaseq` to return the location of the count files that were produced by the RNA-Seq expression pipeline.\n",
    "\n",
    "In this section of the tutorial we will cover:\n",
    "\n",
    "  * using `pf rnaseq` to get count files generated by the RNA-Seq expression pipeline\n",
    "  * filtering `pf rnaseq` results by mapper and reference \n",
    "  * using `pf rnaseq` to symlink count files generated by the RNA-Seq expression pipeline\n",
    "  * using `pf rnaseq` to summarise the relationship between lane, sample and file names\n",
    "  * using `pf rnaseq` to get mapping statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First, let's tell the system the location of our tutorial configuration file.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "export PF_CONFIG_FILE=$PWD/data/pathfind.conf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Let's take a look at the `pf rnaseq` usage.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now, let's get the RNA-Seq expression pipeline results for lane 8479_4#17.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By default, this returns the locations of the expression count files which were produced by the RNA-Seq expression pipeline. \n",
    "\n",
    "For human and mouse data, there will also be featurecount files available which are produced by [featureCounts](http://bioinf.wehi.edu.au/featureCounts/).  You can get these files using the `--filetype` or `-f` option."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -f featurecounts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The mapping pipeline is run before the RNA-Seq expression pipeline. A quick way to get information about which mapper and reference were used by the mapping pipeline is to use the `--details` or `-d` option.\n",
    "\n",
    "**Let's get the mapping details for lane 8479_4#17.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -d"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we can see that \"bwa\" and \"tophat\" were used as the **mapper** and \"Mus_musculus_mm9\" and \"Mus_musculus_mm10\" were used as **references**. \n",
    "\n",
    "You can request the RNA-Seq expression pipeline be run more than once using different mappers or reference. To filter the output by mapper we can use the `--mapper` or `-M` option and the `--reference` or `-R` option to filter by reference.\n",
    "\n",
    "**Let's look for RNA-Seq expression pipeline results for lane 8479_4#17 which used the mapper \"bwa\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -M bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Let's look for RNA-Seq expression pipeline results for lane 8479_4#17 which used the reference \"Mus_musculus_mm9\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -R \"Mus_musculus_mm9\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can symlink our count files into a directory using the `--symlink` or `-l` option.\n",
    "\n",
    "**Let's symlink our count files for lane 8479_4#17 to \"my_count_files\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -l my_count_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ls my_count_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You may want to know the relationship between the lane name, sample name and file name.  We can get this relationship using the `--summary` or `-S` option."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -S"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generates a new file called \"8479_4_17.rnaseqfind_summary.tsv\" which contains the output from `pf info` for this lane (Lane, Sample, Supplier_Name, Public_Name, Strain), the filename that is symlinked (Filename) and the full location to that file (File_Path). This may also be useful as a starting point for your [DEAGO targets file](https://github.com/sanger-pathogens/Bio-Deago/blob/master/user_guide/Input-files.ipynb)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cat 8479_4_17.rnaseqfind_summary.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also get some mapping statistics from our RNA-Seq results using the `--stats` or `-s` option.\n",
    "\n",
    "**Let's get some mapping statistics for lane 8479_4#17.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf rnaseq -t lane -i 8479_4#17 -s "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generated a new file called \"8479_4_17.rnaseqfind_stats.csv\" which contains our mapping statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cat 8479_4_17.rnaseqfind_stats.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that there are two rows for lane 8479_4#17. This is because the mapping pipeline was run twice on this lane using different references and mappers."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Questions\n",
    "\n",
    "**Q1: How many count files are returned by default for run 8479_4?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q2: Which mapper was used with the mapping pipeline for lane 8479_4#18?**  \n",
    "_Hint: the mapper is in the 3rd column of the details_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q3: Which reference was used with the mapping pipeline for lane 8479_4#18?**  \n",
    "_Hint: the reference is in the 2nd column of the details_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q4: What is the location or path of the featurecounts file for lane 8479_4#18?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q5: Which of the lanes in run 8479_4 has the lowest percentage of mapped reads?**  \n",
    "_Hint: you can use `awk` to print out column 2 (lane name) and column 12 (mapped %) of your run statistics_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q6: What is the sample name and symlinked file name associated with lane 8479_4#18?**  \n",
    "_Hint: you might want to summarise the results_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What's next?\n",
    "\n",
    "For a quick recap of how to get annotation pipeline results, head back to [annotation pipeline results](annotation-pipeline-results.ipynb).\n",
    "\n",
    "Otherwise, let's move on to how to [find a reference](finding-a-reference.ipynb)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
