{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation pipeline results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction\n",
    "\n",
    "When your sample data is in the Pathogen Informatics databases, it becomes available to the automated analysis pipelines. After the analysis pipelines have been requested and run, you can use the `pf` scripts to return the results of each of the automated analysis pipelines.\n",
    "\n",
    "The [annotation pipeline](http://mediawiki.internal.sanger.ac.uk/index.php/Pathogen_Informatics_Automated_Annotation_Pipeline) prepares genomes for submission to EMBL/GenBank in a standardised manner, with all the annotation files tracked centrally and available with the `pf annotation` command. \n",
    "\n",
    "In this section of the tutorial we will cover:\n",
    "\n",
    "  * using `pf annotation` to get GFF files generated by the annotation pipeline\n",
    "  * filtering `pf annotation` results by assembler \n",
    "  * using `pf annotation` to symlink files generated by the annotation pipeline \n",
    "  * using `pf annotation` to get annotation statistics\n",
    "  * using `pf annotation` to check whether a gene is found in the GFF files generated by the annotation pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 9"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First, let's tell the system the location of our tutorial configuration file.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "export PF_CONFIG_FILE=$PWD/data/pathfind.conf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Let's take a look at the `pf annotation` usage.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now, let's get the annotation pipeline results for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This returns the locations of the GFF files which were produced by the annotation pipeline. Two paths were returned.  This is because the assembly pipeline was run twice on this lane with two different assemblers: SPAdes and Velvet. Now, look closely at the annotation file paths. Did you notice that each annotation file that's returned is within an assembly directory? This is because your annotations and assemblies are linked.\n",
    "\n",
    "We can filter the returned results, looking only for those results associated with a particular assembly program e.g. SPAdes, Velvet or IVA.  To do this, we can use the `--program` or `-P` option as we did with `pf assembly`.\n",
    "\n",
    "**Let's get all annotation pipeline results for run 5477_6#1 which were generated using \"spades\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -P spades"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "That leaves us with just one annotation file in the \"spades_assembly\" directory.\n",
    "\n",
    "Several different file formats are generated by the annotation pipeline. By default, `pf annotation` returns the GFF file produced by the annotation pipeline. We can also ask `pf annotation` to return a different filetype e.g. the protein FASTA file of the translated CDS sequences.  To do this we use the `--filetype` or `-f` option.\n",
    "\n",
    "**Let's look for the protein FASTA file of the translated CDS sequences for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -P spades -f faa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we got files returned with the extension \".faa\" instead of the default \".gff\" files. We can now symlink these FASTA files into a directory using the `--symlink` or `-l` option.\n",
    "\n",
    "**Let's symlink our protein FASTA files for our SPAdes assembly from lane 5477_6#1 to \"my_protein_files\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -P spades \\\n",
    "    -f faa -l my_protein_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ls my_protein_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also get some statistics from our annotation results using the `--stats` or `-s` option.\n",
    "\n",
    "**Let's get some annotation statistics for our SPAdes assembly from lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -P spades -s"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generated a new file called \"5477_6_1.annotationfind_stats.csv\" which contains our annotation statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cat 5477_6_1.annotationfind_stats.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, we can check to see if a gene is present in our sample using the `--gene` or `-g` option.\n",
    "\n",
    "**Let's see if any of our assemblies for lane 5477_6#1 contain the gene \"_dnaG_\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -g dnaG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we can see that both of our samples contain the _dnaG_ gene.  To check this, we can can use `grep`.\n",
    "\n",
    "**Use `grep` to search for \"_dnaG_\" in the SPAdes annotation for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf annotation -t lane -i 5477_6#1 -P spades | xargs grep dnaG "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Questions\n",
    "\n",
    "**Q1: How many GFF files are returned by default for lane 10018_1#1?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q2: What is the location of the annotation for the SPAdes assembly of lane 10018_1#1?**  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q3: What is the location of the translated CDS sequence file for the SPAdes assembly of lane 10018_1#1?**  \n",
    "_Hint: think about file extensions_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q4: How many of the assemblies for run 5477_6 contain the gene \"_dnaG_\"?**  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What's next?\n",
    "\n",
    "For a quick recap of how to get QC pipeline results, head back to [assembly pipeline results](assembly-pipeline-results.ipynb).\n",
    "\n",
    "Otherwise, let's move on to how to get your [RNA-Seq expression pipeline results](rnaseq-pipeline-results.ipynb)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
