{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping pipeline results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction\n",
    "\n",
    "When your sample data is in the Pathogen Informatics databases, it becomes available to the automated analysis pipelines. After the analysis pipelines have been requested and run, you can use the `pf` scripts to return the results of each of the automated analysis pipelines.\n",
    "\n",
    "The [mapping pipeline](http://mediawiki.internal.sanger.ac.uk/index.php/Pathogen_Informatics_Mapping_Pipeline) maps your raw sequence reads to a reference that you selected. We can use `pf map` to return the location of the BAM files that were produced by the mapping pipeline.\n",
    "\n",
    "In this section of the tutorial we will cover:\n",
    "\n",
    "  * using `pf map` to get BAM files generated by the mapping pipeline\n",
    "  * filtering `pf map` results by mapper and reference \n",
    "  * using `pf map` to symlink BAM files generated by the mapping pipeline \n",
    "  * using `pf map` to get mapping statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First, let's tell the system the location of our tutorial configuration file.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "export PF_CONFIG_FILE=$PWD/data/pathfind.conf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Let's take a look at the `pf map` usage.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now, let's get the mapping pipeline results for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This returns the locations of the BAM files which were produced by the mapping pipeline.\n",
    "\n",
    "A quick way to get information about which mapper and reference were used by the mapping pipeline is to use the `--details` or `-d` option.\n",
    "\n",
    "**Let's get the mapping details for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 -d"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we can see that \"smalt\" was use as the **mapper** and \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" was used as the **reference**. \n",
    "\n",
    "You can request the mapping pipeline be run more than once using different mappers or reference. To filter the output by mapper we can use the `--mapper` or `-M` option and the `--reference` or `-R` option to filter by reference.\n",
    "\n",
    "**Let's look for mapping pipeline results for lane 5477_6#1 which used the mapper \"smalt\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 -M smalt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we got the same results as before. But, what if we try looking for results produced by a different mapper?\n",
    "\n",
    "**Let's look for mapping pipeline results for lane 5477_6#1 which used the mapper \"bwa\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 -M bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This gave us \"No data found\" as BWA hasn't been run on this lane.\n",
    "\n",
    "**Let's look for mapping pipeline results for lane 5477_6#1 which used the reference \"Streptococcus_pneumoniae_Taiwan19F-14_v1\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 \\\n",
    "    -R \"Streptococcus_pneumoniae_Taiwan19F-14_v1\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that we only get one BAM file (.bam) and its index (.bai) returned.  This is because the mapping pipeline has been run twice on this lane, once using the reference \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" and once with \"Streptococcus_pneumoniae_ATCC_700669_v1\".\n",
    "\n",
    "We could then symlink the BAM files into a directory using the `--symlink` or `-l` option.\n",
    "\n",
    "**Let's symlink our BAM files for lane 5477_6#1 to \"my_bam_files\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 \\\n",
    "    -R \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" \\\n",
    "    -l my_bam_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ls my_bam_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also get some statistics from our mapping results using the `--stats` or `-s` option.\n",
    "\n",
    "**Let's get some mapping statistics for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf map -t lane -i 5477_6#1 -s "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generated a new file called \"5477_6_1.mapping_stats.csv\" which contains our mapping statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cat 5477_6_1.mapping_stats.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that there are two rows for lane 5477_6#1. This is because the mapping pipeline was run twice on this lane using different references."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Questions\n",
    "\n",
    "**Q1: How many BAM files are returned by default for lane 5477_6#10?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q2: Which mappers have been used with the mapping pipeline for lane 5477_6#10?**  \n",
    "_Hint: the mapper is in the 3rd column of the details_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q3: Which references have been used with the mapping pipeline for lane 5477_6#10?**  \n",
    "_Hint: the reference is in the 2nd column of the details_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q4: What percentage of the reads from lane 5477_6#10 were mapped to \"Streptococcus_pneumoniae_OXC141_v1\"?**  \n",
    "_Hint: you can use `awk` to filter the statistics file by column 8 (reference) (make sure you set -F',' as the stats are comma-delimited!)_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What's next?\n",
    "\n",
    "For a quick recap of how to get QC pipeline results, head back to [QC pipeline results](qc-pipeline-results.ipynb).\n",
    "\n",
    "Otherwise, let's move on to how to get your [SNP pipeline results](snp-pipeline-results.ipynb)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
