{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SNP pipeline results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction\n",
    "\n",
    "When your sample data is in the Pathogen Informatics databases, it becomes available to the automated analysis pipelines. After the analysis pipelines have been requested and run, you can use the `pf` scripts to return the results of each of the automated analysis pipelines.\n",
    "\n",
    "The [SNP calling pipeline](http://mediawiki.internal.sanger.ac.uk/index.php/Pathogen_Informatics_SNP_Calling_Pipeline) is composed of two parts, SNP calling and pseudogenome construction. We can use `pf snp` to return the location of the variant calling format (VCF) files that were produced by the SNP calling pipeline.\n",
    "\n",
    "In this section of the tutorial we will cover:\n",
    "\n",
    "  * using `pf snp` to get VCF files generated by the SNP calling pipeline\n",
    "  * filtering `pf snp` results by mapper and reference \n",
    "  * using `pf snp` to symlink VCF files generated by the SNP calling pipeline \n",
    "  * using `pf snp` to generate a pseudogenome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First, let's tell the system the location of our tutorial configuration file.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "export PF_CONFIG_FILE=$PWD/data/pathfind.conf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Let's take a look at the `pf snp` usage.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now, let's get the SNP calling pipeline results for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This returns the locations of the gzipped VCF files (.vcf.gz) and their indices (.vcf.gz.tbi) which were produced by the SNP calling pipeline.\n",
    "\n",
    "The mapping pipeline is run before the SNP calling pipeline. A quick way to get information about which mapper and reference were used by the mapping pipeline is to use the `--details` or `-d` option.\n",
    "\n",
    "**Let's get the SNP calling details for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 -d"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we can see that \"smalt\" was use as the **mapper** for both the \"Streptococcus_pneumoniae_ATCC_700669_v1\" and \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" **references**. \n",
    "\n",
    "You can request the SNP calling pipeline be run more than once using different mappers or reference. To filter the output by mapper we can use the `--mapper` or `-M` option and the `--reference` or `-R` option to filter by reference.\n",
    "\n",
    "**Let's look for SNP calling pipeline results for lane 5477_6#1 which used the mapper \"smalt\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 -M smalt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we got the same results as before. But, what if we try looking for results produced by a different mapper?\n",
    "\n",
    "**Let's look for SNP calling pipeline results for lane 5477_6#1 which used the mapper \"bwa\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 -M bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This gave us \"No data found\" as BWA mapping hasn't been run on this lane.\n",
    "\n",
    "**Let's look for SNP calling pipeline results for lane 5477_6#1 which used the reference \"Streptococcus_pneumoniae_Taiwan19F-14_v1\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 \\\n",
    "    -R \"Streptococcus_pneumoniae_Taiwan19F-14_v1\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that we only get one VCF file (.bam) and its index (.tbi) returned.  This is because the SNP calling pipeline has been run twice on this lane, once using the reference \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" and once with \"Streptococcus_pneumoniae_ATCC_700669_v1\".\n",
    "\n",
    "We could then symlink the gzipped VCF files into a directory using the `--symlink` or `-l` option.\n",
    "\n",
    "**Let's symlink our gzipped VCF files for lane 5477_6#1 to \"my_vcf_files\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 \\\n",
    "    -R \"Streptococcus_pneumoniae_Taiwan19F-14_v1\" \\\n",
    "    -l my_vcf_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, we can generate the **pseudogenome**, a custom genome that incorporates filtered variants with respect to the reference. To generate the pseudogenome, we use the `--pseudogenome` or `-p` option.\n",
    "\n",
    "**Let's get the pseudogenomes for lane 5477_6#1.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pf snp -t lane -i 5477_6#1 -p"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that there are two pseudogenome files for lane 5477_6#1. This is because the SNP calling has been run twice."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Questions\n",
    "\n",
    "**Q1: How many lanes from run 10018_1 has the SNP calling pipeline been completed on?**  \n",
    "_Hint: is there another `pf` command you can use to check the status of run?_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q2: How many gzipped VCF files are returned by default for lane 10018_1#20?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q3: Which mapper and reference was used by the SNP calling pipeline for lane 10018_1#20?**  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q4: Generate the pseudogenome for lane 10018_1#20 excluding the reference.**  \n",
    "_Hint: look at the `pf snp` usage_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Q5: Symlink the gzipped VCF files generated by the SNP calling pipeline for run 10018_1 to a new directory called \"10010_1_vcfs\".**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Enter your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What's next?\n",
    "\n",
    "For a quick recap of how to get QC pipeline results, head back to [mapping pipeline results](mapping-pipeline-results.ipynb).\n",
    "\n",
    "Otherwise, let's move on to how to get your [assembly pipeline results](assembly-pipeline-results.ipynb)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
